### ISSUE TITLE
Does not work with DTDs

### ISSUE THREAD

**Issue by boris-petrov** [highsource/jaxb2-annotate-plugin#44]
I've been trying to get this to work when using a DTD. As far as I understand, the "standard" binding mechanism doesn't work and one has to use `xml-java-binding-schema`. I tried the following two but they don't work:

```xml
<?xml version="1.0"?>
<xml-java-binding-schema xmlns:annox="http://annox.dev.java.net">
    <element name="foo" type="class">
        <annox:annotate target="class">@java.lang.SuppressWarnings("all")</annox:annotate>
    </element>
</xml-java-binding-schema>
```

```xml
<?xml version="1.0"?>
<xml-java-binding-schema xmlns:annox="http://annox.dev.java.net">
    <annox:annotate target="class">@java.lang.SuppressWarnings("all")</annox:annotate>
</xml-java-binding-schema>
```

The first one does nothing and the second one gives an error: `compiler was unable to honor this annox:annotate customization. It is attached to a wrong place, or its inconsistent with other bindings.`

Is it possible at all to do what I'm trying to do?

**Comment by highsource** [comment]
I think DTD bindings do not allow customizations.

What you could try is to use the `customizations` plugin to attach customizations to classes or properties. It's not documented, but here's a test project:

https://github.com/highsource/jaxb2-basics/tree/37049a043960fd0bb657ce6667424456057fde4d/tests/issues

It loads/attaches customizations from XML files from the specified directory:

https://github.com/highsource/jaxb2-basics/blob/37049a043960fd0bb657ce6667424456057fde4d/tests/issues/src/main/resources/org/jvnet/jaxb2_commons/tests/issues/IssueJIIB38Type.xml

Basically `org/jvnet/jaxb2_commons/tests/issues/IssueJIIB38Type.xml` gets attached to `the generated `org.jvnet.jaxb2_commons.tests.issues.IssueJIIB38Type`.

No guarantees this will work though.

**Comment by boris-petrov** [comment]
I see, thanks for the answer. I think this is getting way too complicated and probably I'll do a regex replace after xjc. A couple of things:

`I think DTD bindings do not allow customizations.`
One could add a `<xjc:serializable/>` under `xml-java-binding-schema` and that would make the generated classes serializable. I'm not sure that's what you mean but that is one "customization" that could be done with DTDs.

Also, version 1.0.4 of your plugin is not on Maven. :)

Thanks for the support!

**Comment by highsource** [comment]
`xjc:serializable` is an XJC built-in customization so it is somewhat "baked into" the XJC code.
What you need for `jaxb2-annotate-plugin` are so-called "vendor extensions", basically non-JAXB and non-XJC customizations. These are handled differently so I'm not surprized DTD bindings don't allow them.

**Comment by boris-petrov** [comment]
Ok, thank you very much for the answers! I'll leave you to close the issue if you decide so. It's a pity there's nothing that can be done. Thank you again!
